subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java

```diff
@@ -129,6 +131,11 @@ public void apply(final Project project) {
         configureBuildNeeded(project);
         configureBuildDependents(project);
         bridgeToSoftwareModelIfNecessary(projectInternal);
+        configureToolchainsTask(project);
+    }
+
+    private void configureToolchainsTask(Project project) {
+        project.getTasks().register("javaToolchains",  ShowToolchainsTask.class, new ShowToolchainsTaskAction(project.getDisplayName()));
```

Comment:
:o: Can the toolchains be _different_ based on the project? I felt like "nope", but maybe I misunderstood. If it's not the case then it shouldn't be a task created on each project.

Reply:
As discussed, solved.